src/TvMaze.Api.Client/Endpoints/Search/SearchEndpoint.cs

```diff
@@ -26,14 +26,14 @@ public Task<IEnumerable<ShowSearchResult>> ShowSearchAsync(string query)
         }
 
         /// <inheritdoc />
-        public Task<ShowSearchResult> ShowSingleSearchAsync(string query)
+        public Task<Show> ShowSingleSearchAsync(string query, ShowEmbeddingFlags embeddings = ShowEmbeddingFlags.None)
         {
             if (string.IsNullOrWhiteSpace(query))
             {
                 throw new ArgumentNullException(nameof(query));
             }
 
-            return _httpClient.GetAsync<ShowSearchResult>($"singlesearch/shows?q={query}");
+            return _httpClient.GetAsync<Show>(ShowEmbeddings.AddQueryStringToUrl($"singlesearch/shows?q={query}", embeddings));
```

Comment:
I was looking at Flurl's URL builder and it looks really nice. I think we can move the query string building logic to the methods by using the URL builder.

I haven't tested the code below and it's not exact implementation but maybe we can try to do something similar?

For example, we could have a dictionary to map the flags to the query string values:
```c#
var queryStringValues = new Dictionary<ShowEmbeddingFlags, string>
                {
                    { ShowEmbeddingFlags.Episodes, "episodes" },
                    { ShowEmbeddingFlags.Cast, "cast" },
                    { ShowEmbeddingFlags.PreviousEpisode, "previousepisode" },
                    { ShowEmbeddingFlags.NextEpisode, "nextepisode" },
                };
```

We can get a list of all the selected flags:
```c#
var selectedFlags = Enum.GetValues(typeof(ShowEmbeddingFlags))
                    .OfType<ShowEmbeddingFlags>()
                    .Where(flag => flag != ShowEmbeddingFlags.None && embeddings.HasFlag(flag));
```

Then we can add the query string values by using Flurl's URL builder:
```c#
var endpointUrl = $"shows/{showId}".SetQueryParam("embed[]", new[]
                {
                    selectedFlags.Select(flag => queryStringValues[flag])
                });
```

What do you think about this approach?

Reply:
Good idea, I only tried with the .Net library in didn't find a way to have a duplicated query param, but this approach seems to work